projects/slap_manipulation/scripts/eval_peract_agent.py

```diff
@@ -0,0 +1,80 @@
+import click
+import numpy as np
+import rospy
+import trimesh.transformations as tra
+from slap_manipulation.agents.peract_agent import PeractAgent
+from slap_manipulation.env.general_language_env import GeneralLanguageEnv
+
+from home_robot.core.interfaces import ContinuousEndEffectorAction
+from home_robot_hw.utils.config import load_slap_config
+
+
+@click.command()
+@click.option("--test-pick", default=False, is_flag=True)
+@click.option("--dry-run", default=False, is_flag=True)
+# @click.option("--testing/--no-testing", default=False, is_flag=True)
+@click.option("--object", default="cup")
+@click.option("--task-id", default=0)
+@click.option(
+    "--cat-map-file", default="projects/stretch_ovmm/configs/example_cat_map.json"
+)
+def main(task_id, cat_map_file, test_pick=False, dry_run=False, **kwargs):
+    config = load_slap_config(
+        visualize=True,
+        config_path="projects/slap_manipulation/configs/language_agent.yaml",
+        **kwargs
+    )
+    rospy.init_node("eval_peract")
+    agent = PeractAgent(config, task_id=task_id)
+    agent.load_models()
+
+    env = GeneralLanguageEnv(
+        config=config,
+        test_grasping=test_pick,
+        dry_run=dry_run,
+        # segmentation_method="detic",
+        cat_map_file=cat_map_file,
+    )
+
+    env.reset()
+    agent.reset()
+
+    goal_info = agent.get_goal_info()
+    env._set_goal(goal_info)
+    if not env.robot.in_manipulation_mode():
+        env._switch_to_manip_mode(grasp_only=True, pre_demo_pose=True)
+    res = input("Press Y/y to close the gripper")
+    if res == "y" or res == "Y":
+        env._handle_gripper_action(1)
+        # env.robot.manip.close_gripper()
+    else:
+        env._handle_gripper_action(-1)
+        # env.robot.manip.open_gripper()
+    rospy.sleep(3.0)
```

Comment:
why

Reply:
Otherwise starts moving without finishing opening or closing of gripper.